### PR TITLE
feat(terra): threaded PRE_STEPS into the tests stage (Azure DevOps)

### PR DESCRIPTION
**Pull request by rios0rios0**
## Summary
The `test:all` job in `azure-devops/terra/stages/30-tests/terra.yaml` runs `terraform init` per module with a `tests/` directory. Consumers whose modules pin `source = \"git@...\"` references (e.g. shared `terraform-modules` repos on Azure DevOps) need the same SSH setup the SAST tools already receive via `PRE_STEPS` on the `20-security` stage. Without this hook, tier 1 (``terraform test``) fails from the tests job with ``ssh: Could not resolve hostname`` while ``20-security`` resolves the same references cleanly.

This PR:

- adds a ``PRE_STEPS`` parameter to ``azure-devops/terra/stages/30-tests/terra.yaml``.
- injects the consumer-provided step list between the ``terra`` CLI install abstract and the ``test-all/run.sh`` invocation — exactly where SSH setup needs to run.
- threads the same ``PRE_STEPS`` value from the top-level ``azure-devops/terra/terra.yaml`` aggregator so consumers maintain a single list that covers both the security stage and the tests stage.

## Test plan
- [x] ``yamllint`` clean (no config drift)
- [x] consumer repo (``ZestSecurity/shared-toolbox``) threads ``PRE_STEPS: [setup-modules-repo.yaml@terra]`` and tier 1 goes from 23/30 cases (SSH-failing init) to 30/30 green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: azure-devops/terra/stages/30-tests/terra.yaml [hunk 1]
```diff
@@ -25,6 +36,16 @@ stages:
...
 
+          # Run consumer-provided pre-steps (e.g. configuring SSH for
+          # private-git module sources) before `test-all/run.sh` calls
+          # `terraform init` per module. Without this hook, modules that
+          # reference `source = "git@..."` in their bodies fail init from
+          # the tests job with `ssh: Could not resolve hostname`. The
+          # sibling `20-security` stage threads the same parameter into
+          # its SAST tools — consumers already maintain one `PRE_STEPS`
+          # list; we just reuse it here.
+          - ${{ parameters.PRE_STEPS }}
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply azure-devops/terra/stages/30-tests/terra.yaml]: Addressed in `ce435c6`. Switched to the `${{ each step in parameters.PRE_STEPS }}: - ${{ step }}` splice pattern to match `azure-devops/global/stages/20-security/{docker,trivy}.yaml` and avoid the empty-list edge case.

File: azure-devops/terra/stages/30-tests/terra.yaml [hunk 2]
```diff
@@ -1,3 +1,14 @@
+parameters:
+  # Steps injected before the Terraform `test-all` runner invokes
+  # `terraform init` + `terraform test` on each module. Consumers whose
+  # modules pin private `source = "git@..."` paths pass the setup steps
+  # that configure SSH so that `terraform init` can resolve those
+  # references from the tests job (where the management-stage SSH
+  # setup is not available).
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply azure-devops/terra/stages/30-tests/terra.yaml]: Addressed in `ce435c6`. Reworded the doc comment to reference the sibling `20-security` `PRE_STEPS` hook instead of a non-existent management-stage SSH setup.

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ Exceptions are acceptable depending on the circumstances (critical bug fixes tha
 
 ### Added
 
+- added `PRE_STEPS` parameter to `azure-devops/terra/stages/30-tests/terra.yaml` (the `test:all` job) and wired the top-level `azure-devops/terra/terra.yaml` to thread the same consumer-provided list through. Previously `PRE_STEPS` only reached the `20-security` stage — the tests job had no hook, so consumers whose modules pin private `source = "git@..."` references hit `ssh: Could not resolve hostname` inside `terraform init` during tier 1 (`terraform test`) while the SAST tools in the security stage resolved the same references cleanly. With this change, a single `PRE_STEPS` list (typically `setup-modules-repo.yaml@terra` in Azure DevOps consumers) covers both stages, keeping the consumer contract symmetric with how the delivery/management stages already handle it.
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply CHANGELOG.md]: Addressed in `ce435c6`. Reworded the entry to describe symmetry between the `20-security` and `30-tests` stages, which are the two stages that actually accept `PRE_STEPS`.
